src/server.cpp

```diff
@@ -2475,33 +2475,41 @@ void Server::sendRequestedMedia(session_t peer_id,
 
 void Server::sendDetachedInventory(const std::string &name, session_t peer_id)
 {
-	if(m_detached_inventories.count(name) == 0) {
-		errorstream<<FUNCTION_NAME<<": \""<<name<<"\" not found"<<std::endl;
-		return;
-	}
-	Inventory *inv = m_detached_inventories[name];
-	std::ostringstream os(std::ios_base::binary);
+	const auto &inv_it = m_detached_inventories.find(name);
+	const auto &player_it = m_detached_inventories_player.find(name);
 
-	os << serializeString(name);
-	inv->serialize(os);
+	if (player_it == m_detached_inventories_player.end() ||
+			player_it->second.empty()) {
+		// ok. no restriction
+	} else {
+		RemotePlayer *p = m_env->getPlayer(player_it->second.c_str());
+		if (!p)
+			return; // Player is offline
 
-	// Make data buffer
-	std::string s = os.str();
+		if (peer_id != PEER_ID_INEXISTENT && peer_id != p->getPeerId())
+			return; // Send to nobody
```

Comment:
this is hard to grasp on first sight
<pre>
			return; // Caller requested send to a different player, so don't send.
</pre>